File: 2022/Day10/Puzzel1.cpp
#include <fstream>
#include <iostream>
#include <string>
#include "StringUtils.hpp"
#include <vector>

int main()
{
    std::ifstream inputFile("input.txt");
    std::string temp;
    int registerValue = 1;
    int cycle = 0;
    int total = 0;
    while( std::getline( inputFile, temp ) )
    {
        std::vector<std::string> line = StringUtils::splitString( temp, ' ' );
        if( line[0] == "addx" )
        {
            for( unsigned int i = 0; i<2; i++ )
            {
                cycle++;
                if( ( cycle - 20 ) % 40 == 0)
                {
                    std::cout << "Cycle: " << cycle << " Register value: " << registerValue << '\n';
                    total += cycle * registerValue;
                }
            }
            registerValue += std::stoi( line[1] );
        }
        else
        {
            cycle++;
            if( ( cycle - 20 ) % 40 == 0)
            {
                std::cout << "Cycle: " << cycle << " Register value: " << registerValue << '\n';
                total += cycle * registerValue;
            }
        }    
    }
    std::cout << "Total: " << total << '\n';
}

File: 2022/Day10/Puzzel1.py
input = []
with open("input.txt", "r") as inputFile:
    input = inputFile.readlines()

registerValue = 1
cycle = 0
total = 0
    
for line in input:
    line = line.split(" ")
    if( line[0] == "addx" ):
        for i in range(2):
            cycle += 1
            if( ( cycle - 20 ) % 40 == 0):
                print("Cycle: " + str(cycle) + " Register value: " + str(registerValue))
                total += cycle * registerValue
        registerValue += int(line[1])
    else:
        cycle += 1
        if( ( cycle - 20 ) % 40 == 0):
                print("Cycle: " + str(cycle) + " Register value: " + str(registerValue))
                total += cycle * registerValue

print(f"Total: {total}\n")
